integration_test/third_party_apps_test.go

```diff
@@ -222,6 +253,102 @@ func runLoggingTestCases(ctx context.Context, logger *logging.DirectoryLogger, v
 	return err
 }
 
+func runMetricsTestCases(ctx context.Context, logger *logging.DirectoryLogger, vm *gce.VM, testCaseBytes []byte) error {
+	var entries []expectedMetricEntry
+	var requiredEntries []expectedMetricEntry
+	err := yaml.UnmarshalStrict(testCaseBytes, &entries)
+	if err != nil {
+		return fmt.Errorf("could not unmarshal contents of expected_metrics.yaml: %v", err)
+	}
+	if len(entries) == 0 {
+		logger.ToMainLog().Printf("Skipping metrics test: expected_metrics.yaml is empty")
+		return nil
+	}
+	representativeCount := 0
+	for _, entry := range entries {
+		if entry.IsRepresentative {
+			representativeCount += 1
+		}
+	}
+	if representativeCount != 1 {
+		return fmt.Errorf("There must be exactly one metric with is_representatve: true. Found %v.", representativeCount)
+	}
+	logger.ToMainLog().Printf("Parsed expected_metrics.yaml: %+v", entries)
+	for _, entry := range entries {
+		if entry.Optional {
+			logger.ToMainLog().Printf("Skipping expected metric %s because it is marked optional", entry.Type)
+		} else {
+			requiredEntries = append(requiredEntries, entry)
+		}
+	}
+	c := make(chan error, len(requiredEntries))
+	for _, entry := range requiredEntries {
+		entry := entry
+		go func() {
+			c <- assertMetric(ctx, logger, vm, entry)
```

Comment:
as mentioned over IM, I think this will blow us out of monitoring API quota in a hurry when testing cross-platform. Especially if testing all apps on all platforms. I proposed checking for the representative metric first, then a sleep, then checking for the rest of the metrics in parallel. That's what I did in ops_agent_test when I had a very similar issue.

Reply:
This should be implemented now.